### PR TITLE
rename example dir name and change workflow in examples

### PR DESCRIPTION
**Pull request by mengqiy**

### CODE REVIEW

File: examples/ensure-name-substring/ensure-name-substring-prefix/README.md [hunk 1]
```diff
@@ -5,6 +5,14 @@
 This example demonstrates how to declaratively run the [`ensure-name-substring`]
 function to prepend prefix in the resource names.
 
+### Fetch the example package
+
+Get the example package by running the following commands:
+
+```shell
+$ kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-catalog.git/examples/ensure-name-substring/ensure-name-substring-prefix
```
Comment (frankfarzan) [inline examples/ensure-name-substring/ensure-name-substring-prefix/README.md:13]: you don't need the intermediate directory any more:

`examples/<function-name>-<example-name>`

File: examples/apply-setters/apply-setters-simple/README.md [hunk 2]
```diff
@@ -8,6 +8,14 @@ with setter references.
 In this example, we will see how to apply desired setter values to the 
 resource fields parameterized by `kpt-set` comments.
 
+### Fetch the example package
+
+Get the example package by running the following commands:
+
+```shell
+$ kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-catalog.git/examples/apply-setters/apply-setters-simple
```
Comment (frankfarzan) [inline examples/apply-setters/apply-setters-simple/README.md:16]: Can you do an audit of kpt.dev content that refer to example packages in the catalog? Also at some point, need to release these packages and uprev the kpt.dev content to use them.

File: examples/ensure-name-substring/ensure-name-substring-prefix/README.md (hunk 1 above)
Comment (mengqiy) [reply examples/ensure-name-substring/ensure-name-substring-prefix/README.md:13]: We can remove the intermediate directory, but I need to modify the site deployment script.

File: examples/apply-setters/apply-setters-simple/README.md (hunk 2 above)
Comment (mengqiy) [reply examples/apply-setters/apply-setters-simple/README.md:16]: Will create a PR in kpt repo soon.

Comment (mengqiy) [reply examples/apply-setters/apply-setters-simple/README.md:16]: https://github.com/GoogleContainerTools/kpt/pull/2247

File: examples/ensure-name-substring/ensure-name-substring-prefix/README.md (hunk 1 above)
Comment (mengqiy) [reply examples/ensure-name-substring/ensure-name-substring-prefix/README.md:13]: @etefera is going to help with it.
It is tracked in https://github.com/GoogleContainerTools/kpt/issues/2250

**Comment by mengqiy** [comment]
@droot PTAL

File: examples/gatekeeper-imperative/README.md [hunk 3]
```diff
@@ -15,21 +23,15 @@ We have a `ConfigMap` in `config-map.yaml` that violates the policy.
...
-
-```shell
-$ kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-catalog.git/examples/gatekeeper/imperative
-```
-
 Run the function with `--results-dir` flag:
 
 ```shell
-$ kpt fn eval --image=gcr.io/kpt-fn/gatekeeper:unstable --results-dir=results
+$ kpt fn eval gatekeeper-imperative --image=gcr.io/kpt-fn/gatekeeper:unstable --results-dir=/tmp
```
Comment (droot) [inline examples/gatekeeper-imperative/README.md:29]: not a big deal. `=` is not needed here. just `--results-dir /tmp` is pleasing to the eyes.

File: examples/gatekeeper-imperative/README.md [hunk 4]
```diff
@@ -15,21 +23,15 @@ We have a `ConfigMap` in `config-map.yaml` that violates the policy.
...
 ```
 
 ### Expected result
 
-Let's take a look at the structured result:
+Let's take a look at the structured result in `/tmp/results.yaml`:
 
 ```yaml
 apiVersion: kpt.dev/v1alpha2
```
Comment (droot) [inline examples/gatekeeper-imperative/README.md:37]: once, we uprev the versions to `v1`, we will have to update these docs, but `find/replace` should be enough.

File: scripts/verify-docs.py [hunk 5]
```diff
@@ -183,10 +223,10 @@ def validate_metadata(metadata, branch, path, fn, examples_list):
     desired_source_url = f'https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/{branch}/{path}'
     if metadata['sourceURL'] != desired_source_url:
         raise Exception(f'{fn}: "sourceURL" should be "{desired_source_url}"')
-    if len(metadata['examplePackageURLs']) != len(examples_list):
-        raise Exception(f"{fn}: examplePackageURLs have {len(metadata['examplePackageURLs'])} examples, but the examples directory has {len(examples_list)}")
+    if len(examples_list) == 0:
+        raise Exception(f'{fn}: there must be at least one example listed in metadata.yaml file')
     for example in examples_list:
-        desired_example_pkg_url = f'https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/{branch}/examples/{fn}/{example}'
+        desired_example_pkg_url = f'https://github.com/GoogleContainerTools/kpt-functions-catalog/tree/{branch}/examples/{example}'
```
Comment (droot) [inline scripts/verify-docs.py:229]: hmmm `python`. This is a decent size script now. We should try to minimize the number of languages used in the project from ease of maintenance perspective.

**Review by droot — APPROVED** [review]

File: examples/gatekeeper-imperative/README.md (hunk 4 above)
Comment (mengqiy) [reply examples/gatekeeper-imperative/README.md:37]: Phani is taking care of it in https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/408

File: scripts/verify-docs.py (hunk 5 above)
Comment (mengqiy) [reply scripts/verify-docs.py:229]: Ack

**Pull request merged**